### PR TITLE
benchalerts: move from separate repo

### LINKED ISSUE #719: monorepo: bring `benchalerts` into this repo
To take advantage of this repo's CI infrastructure (and make possible better future integration tests), we should move https://github.com/conbench/benchalerts into this repo.

### PR DESCRIPTION
**Pull request by austin3dickey**
Closes #719. This PR brings in almost everything from https://github.com/conbench/benchalerts into this repo as another package.

It does not bring the "license check" or "coverage check" CI steps.

I added an Overview section to the README as well.

### CODE REVIEW

**Comment by austin3dickey** [comment]
Many of the integration tests are timing out because of #744 and #745. I am going to disable them in CI for now and create an issue: #748

**Comment by austin3dickey** [comment]
CI is now passing. Will merge to iterate quickly. Feedback is always welcome.

**Review by jgehrcke — COMMENTED** [review]
:rocket:!

File: benchalerts/README.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+A package to facilitate automated alerting based on Conbench data.
+
+## Overview
+
+This package is intended to make the following steps easier in CI. Before these steps,
+it is assumed that an execution environment has performed a run of benchmarks and
+submitted the results to Conbench.
+
+- Hit the Conbench API to understand if there were any:
```
Comment (jgehrcke) [inline benchalerts/README.md:11]: For the future: 

I'd love to see a summary for the method/algorithm applied here so far. (text :-).

File: benchalerts/benchalerts/clients.py [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+import jwt
+import requests
+from requests.adapters import HTTPAdapter
+from urllib3.util.retry import Retry
+
+from .log import fatal_and_log, log
+
+
+class _BaseClient(abc.ABC):
+    """A client to interact with an API.
```
Comment (jgehrcke) [inline benchalerts/benchalerts/clients.py:32]: For the future: with which parts in this repo does this overlap?

(enumeration of code that implements the same or a similar idea)

File: benchalerts/tests/unit_tests/mocked_responses/GET_conbench_benchmarks_run_id_contender_wo_base.json [hunk 3]
```diff
@@ -0,0 +1,114 @@
+{
+    "status_code": 200,
+    "data": [
+        {
+            "batch_id": "some-batch-uuid-1",
```
Comment (jgehrcke) [inline benchalerts/tests/unit_tests/mocked_responses/GET_conbench_benchmarks_run_id_contender_wo_base.json:5]: For future: might need a bit of a plan to update mocked responses once we update API behavior / do integration testing with less mocking.

File: benchalerts/tests/unit_tests/test_parse_conbench.py [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+    actual = benchmarks_with_z_regressions(mock_comparisons)
+    assert len(actual) == expected_len
+
+
+@pytest.mark.parametrize(
+    ["mock_comparisons", "expected_md"],
+    [
+        ("noregressions", "summary_noregressions"),
+        ("regressions", "summary_regressions"),
+        ("no_baseline", "summary_nobaseline"),
```
Comment (jgehrcke) [inline benchalerts/tests/unit_tests/test_parse_conbench.py:146]: :heart:

File: benchalerts/README.md (hunk 1 above)
Comment (austin3dickey) [reply benchalerts/README.md:11]: Great idea! Right now, it basically just hits 
`/api/compare/runs/{baseline}...{contender}/?threshold_z={threshold_z}`
 for each contender run associated with the given commit. All the logic for determining what the baseline run is and how to define the z-score with which to threshold is on the server's side. I can put this in the README.

File: benchalerts/benchalerts/clients.py (hunk 2 above)
Comment (austin3dickey) [reply benchalerts/benchalerts/clients.py:32]: The `benchclients` package was inspired by this code. @alistaire47 and I both needed a python client to hit the Conbench API for `benchadapt` and `benchalerts` respectively, so he created `benchclients` so both packages could depend on the same code.

I still need to do https://github.com/conbench/benchalerts/issues/16 to delete this code and have this package depend on `benchclients` so we can reduce duplication. (I should move that repo's issues to this repo now!)

File: benchalerts/tests/unit_tests/mocked_responses/GET_conbench_benchmarks_run_id_contender_wo_base.json (hunk 3 above)
Comment (austin3dickey) [reply benchalerts/tests/unit_tests/mocked_responses/GET_conbench_benchmarks_run_id_contender_wo_base.json:5]: Absolutely!! I'm hoping we can rely a lot less on brittle mocks once we do #720 some more.

File: benchalerts/benchalerts/clients.py (hunk 2 above)
Comment (austin3dickey) [reply benchalerts/benchalerts/clients.py:32]: okay, moved all the issues, so this is now https://github.com/conbench/conbench/issues/754

File: benchalerts/README.md (hunk 1 above)
Comment (austin3dickey) [reply benchalerts/README.md:11]: #757

**Pull request merged**